src/openms/source/ANALYSIS/QUANTITATION/PeptideAndProteinQuant.cpp

```diff
@@ -391,82 +396,93 @@ namespace OpenMS
 
     for (auto const& pep_q : pep_quant_)
     {
-      String accession = getAccession_(pep_q.second.accessions,
-                                       accession_to_leader);
+      String accession = getAccession_(pep_q.second.accessions, accession_to_leader);
       OPENMS_LOG_DEBUG << "Peptide id mapped to leader: " << accession << endl;
 
       // not enough evidence or mapping to multiple groups
-      if (accession.empty()) continue;
+      if (accession.empty())
+        continue;
 
       // proteotypic peptide
       const String peptide = pep_q.first.toUnmodifiedString();
 
       prot_quant_[accession].psm_count += pep_q.second.psm_count;
 
       // transfer abundances and counts from peptides->protein
-      // summarize abundances and counts between different peptidoforms       
-      for (auto const & sta : pep_q.second.total_abundances)
+      // summarize abundances and counts between different peptidoforms
+      for (auto const& sta : pep_q.second.total_abundances)
       {
         prot_quant_[accession].abundances[peptide][sta.first] += sta.second;
       }
 
-      for (auto const & sta : pep_q.second.total_psm_counts)
+      for (auto const& sta : pep_q.second.total_psm_counts)
       {
         prot_quant_[accession].psm_counts[peptide][sta.first] += sta.second;
       }
     }
 
+    std::string method = param_.getValue("method");
+
+    if (method == "iBAQ")  // TODO: parameter override too late - ProteinQuantifier reads parameters earlier, meaning output file contains incorrect parameters
+    {
+      param_.setValue("top", 0);
+      param_.setValue("average", "sum");
+    }
+
     Size top = param_.getValue("top");
     std::string average = param_.getValue("average");
     bool include_all = param_.getValue("include_all") == "true";
     bool fix_peptides = param_.getValue("consensus:fix_peptides") == "true";
 
-    for (auto & prot_q : prot_quant_)
+    for (auto& prot_q : prot_quant_)
     {
       const ProteinData& pd = prot_q.second;
 
       // calculate PSM counts based on all (!) peptides of a protein (group)
-      for (auto const & pep2sa : pd.psm_counts)
+      for (auto const& pep2sa : pd.psm_counts)
       { // for all peptides of this protein (group)
         const SampleAbundances& sas = pep2sa.second;
-        for (auto const & sa : sas)
+        for (auto const& sa : sas)
         {
           const Size& sample_id = sa.first;
           const Size& psms = sa.second;
-          if (psms > 0) prot_q.second.total_distinct_peptides[sample_id]++; // count this peptide sequence once if observed in sample
-          prot_q.second.total_psm_counts[sample_id] += psms; // count all PSMs of this protein in this sample
+          if (psms > 0)
+            prot_q.second.total_distinct_peptides[sample_id]++; // count this peptide sequence once if observed in sample
+          prot_q.second.total_psm_counts[sample_id] += psms;    // count all PSMs of this protein in this sample
```

Comment:
indentation

Reply:
This one is not from me, are both supposed to be in the if clause?

Reply:
sorry forget it :)